Symfony/CS/Fixer/Contrib/Php4ConstructorFixer.php

```diff
@@ -0,0 +1,167 @@
+
+            // using parent::ParentClassName()
+            $i = $parIndex;
+            while ($i = $tokens->getNextTokenOfKind($i, array(array(T_PAAMAYIM_NEKUDOTAYIM, "::")))) {
+                $p = $tokens->getPrevMeaningfulToken($i);
+                if ($tokens[$p]->equals(array(T_STRING, 'parent'))) {
+                    $i = $tokens->getNextMeaningfulToken($i);
+                    if (!strcasecmp($tokens[$i]->getContent(), $parentClass)) {
+                        // Found parent
+                        $tokens[$i]->setContent('__construct');
+                        break;
+                    }
+                }
+            }
+
+            // using $this->ParentClassName()
+            $i = $parIndex;
+            while ($i = $tokens->getNextTokenOfKind($i, array(array(T_OBJECT_OPERATOR, "->")))) {
+                $p = $tokens->getPrevMeaningfulToken($i);
+                if ($tokens[$p]->equals(array(T_VARIABLE, '$this'))) {
+                    $op = $i;
+                    $i = $tokens->getNextMeaningfulToken($i);
+                    if (!strcasecmp($tokens[$i]->getContent(), $parentClass)) {
+                        // Found parent
+                        $tokens[$p]->setContent('parent');
+                        $tokens[$op]->setContent('::');
+                        $tokens[$i]->setContent('__construct');
+
+                        break;
+                    }
+                }
+            }
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'php4_constructor';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Convert PHP4-style constructors to __construct.';
+    }
+
+    /**
+     * Find a function matching a given name within certain bounds.
+     *
+     * @param  Tokens      $tokens
+     * @param $name
+     * @param $start
+     * @param $end
+     * @return array|false
+     */
+    protected function findFunction(Tokens $tokens, $name, $start, $end)
+    {
+        foreach ($tokens->findGivenKind(T_FUNCTION) as $index => $token) {
+            if ($index < $start || $index > $end) {
+                continue;
+            }
+
+            $funcIndex =  $tokens->getNonWhitespaceSibling($index, 1);
+            $funcName = $tokens[$funcIndex]->getContent();
+
+            if (!strcasecmp($name, $funcName)) {
+                $funcStart = $tokens->getPrevMeaningfulToken($index) + 1;
+                $bodyStart = $tokens->getNextTokenOfKind($funcStart, array('{'));
+                $funcEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $bodyStart);
+                $body = $tokens->generatePartialCode($bodyStart, $funcEnd);
+
+                return array(
+                    'name'  => $funcIndex,
+                    'start' => $funcStart,
+                    'end'   => $funcEnd,
+                    'body'  => $body,
```

Comment:
Please use Symfony's standards as described in CONTRIBUTING file and stop aligning